x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+  RunContext,
+} from '../../../../task_manager/server';
+import { checkRunningSessions } from './check_running_sessions';
+import { CoreSetup, SavedObjectsClient, Logger } from '../../../../../../src/core/server';
+import { BACKGROUND_SESSION_TYPE } from '../../saved_objects';
+
+export const BACKGROUND_SESSIONS_TASK_TYPE = 'bg_monitor';
+export const BACKGROUND_SESSIONS_TASK_ID = `data_enhanced_${BACKGROUND_SESSIONS_TASK_TYPE}`;
+export const MONITOR_INTERVAL = `15s`;
+
+function backgroundSessionRunner(core: CoreSetup, logger: Logger) {
+  return ({ taskInstance }: RunContext) => {
+    return {
+      async run() {
+        const [coreStart] = await core.getStartServices();
+        const internalRepo = coreStart.savedObjects.createInternalRepository([
+          BACKGROUND_SESSION_TYPE,
+        ]);
+        const internalSavedObjectsClient = new SavedObjectsClient(internalRepo);
+        await checkRunningSessions(
+          internalSavedObjectsClient,
+          coreStart.elasticsearch.client.asInternalUser,
+          logger
+        );
+
+        return {
+          runAt: new Date(Date.now() + 10 * 1000),
+          state: {},
+        };
+      },
+    };
+  };
+}
+
+export function registerBackgroundSessionsTask(
+  core: CoreSetup,
+  taskManager: TaskManagerSetupContract,
+  logger: Logger
+) {
+  taskManager.registerTaskDefinitions({
+    [BACKGROUND_SESSIONS_TASK_TYPE]: {
+      title: 'Background Session Monitor',
+      createTaskRunner: backgroundSessionRunner(core, logger),
+    },
+  });
+}
+
+export async function scheduleBackgroundSessionsTasks(
+  taskManager: TaskManagerStartContract,
+  logger: Logger
+) {
+  try {
+    // delete previous task
+    await taskManager.remove(BACKGROUND_SESSIONS_TASK_ID);
+  } catch (e) {} // eslint-disable-line no-empty
+
+  try {
+    await taskManager.ensureScheduled({
+      id: BACKGROUND_SESSIONS_TASK_ID,
+      taskType: BACKGROUND_SESSIONS_TASK_TYPE,
+      schedule: {
+        interval: MONITOR_INTERVAL,
+      },
+      state: {},
+      params: {},
+    });
+
+    logger.debug(`Background search task, scheduled to run`);
+  } catch (e) {
+    logger.debug(`Error scheduling task, received ${e.message}`);
```

Comment:
This should probably detect if the error is something that can be ignored (like a 409 error) and only  use `logger.debug` if it can be ignored. 

Most of the time, the error probably will be a version conflict that can be ignored. If it's anything else, then it should stand out in the logs.

Reply:
As far as I understand, this `logger` call is not related to the content of the task. It's just about scheduling. The error is handled this way in all places scheduling a task.

Regardless, 409's will be caught [here](https://github.com/elastic/kibana/pull/85253/files/1347551ec4af5eb5515c67a42b174d9380bcca0a#diff-c94daef7f046d960efb648177f47e6577c1c5acdefba2ee7b60847f3c50ef9a0R66)

Reply:
I was thinking we could handle scheduling errors, especially a `409` error that could be fired from `ensureScheduled`. But I think you are right: this is the way error handling is done in all the places that schedule a task. It's all good.